### ISSUE TITLE
How to Work with multiple Entity Managers and Connections

### ISSUE THREAD

**Issue by damiarita** [symfony/symfony-docs#10797]
I am not sure if this is an issue with the docs or a bug.

I have created a symfony app with 2 Entity Managers and two connections. I followed the examples of https://symfony.com/doc/current/doctrine/multiple_entity_managers.html I only changed the name of the default connection and the default orm. Instead of calling them `default` I called it `BOE`

It does not look like a wrong thing to do, as I had `default_connection` and `default_entity_manager` keys with value `BOE`.

It turns out that if I do the cache clear with APP_ENV = dev, it works flawlessly. But if i clear the cache for prod, it gives this error:  
> Uncaught PHP Exception Doctrine\Common\Persistence\Mapping\MappingException: "The class '{class name here}' was not found in the chain configured namespaces " at....

**Comment by xabbuh** [comment]
Can you show your Doctrine related config for all environments?

**Comment by damiarita** [comment]
Hi,

Thanks for your reply. This is the content of /config/packages/doctrine.yaml

`parameters:
    # Adds a fallback DATABASE_URL if the env var is not set.
    # This allows you to run cache:warmup even if your
    # environment variables are not available yet.
    # You should not need to change this value.
    env(DATABASE_URL): ''
    env(BOE_DATABASE_DRIVER): 'pdo_mysql'
    env(BOE_DATABASE_VERSION): '5.6.34'
    env(BOE_DATABASE_CHARSET): 'utf8mb4'
    env(BOE_DATABASE_URL): ''

doctrine:
    dbal:
        default_connection: BOE
        connections:
            BOE:
                driver: '%env(BOE_DATABASE_DRIVER)%'
                server_version: '%env(BOE_DATABASE_VERSION)%'
                charset: '%env(BOE_DATABASE_CHARSET)%'
                url: '%env(BOE_DATABASE_URL)%'
            BAS:
                driver: 'pdo_mysql'
                server_version: '5.6.34'
                charset: utf8mb4
                url: '%env(DATABASE_URL)%'
    orm:
        default_entity_manager: BOE
        entity_managers:
            BOE:
                connection: BOE
                mappings:
                    BOE:
                        is_bundle: false
                        type: annotation
                        dir: '%kernel.project_dir%/src/Entity/BOE'
                        prefix: 'App\Entity\BOE'
                        alias: BOE
            BAS:
                connection: BAS
                mappings:
                    BAS:
                        is_bundle: false
                        type: annotation
                        dir: '%kernel.project_dir%/src/Entity/BAS'
                        prefix: 'App\Entity\BAS'
                        alias: BAS
        auto_generate_proxy_classes: '%kernel.debug%'
`

/config/packages/prod/doctrine.yaml

`doctrine:
    orm:
        metadata_cache_driver:
            type: service
            id: doctrine.system_cache_provider
        query_cache_driver:
            type: service
            id: doctrine.system_cache_provider
        result_cache_driver:
            type: service
            id: doctrine.result_cache_provider

services:
    doctrine.result_cache_provider:
        class: Symfony\Component\Cache\DoctrineProvider
        public: false
        arguments:
            - '@doctrine.result_cache_pool'
    doctrine.system_cache_provider:
        class: Symfony\Component\Cache\DoctrineProvider
        public: false
        arguments:
            - '@doctrine.system_cache_pool'

framework:
    cache:
        pools:
            doctrine.result_cache_pool:
                adapter: cache.app
            doctrine.system_cache_pool:
                adapter: cache.system
`


I have no /config/packages/dev/doctrine.yaml or /config/packages/test/doctrine.yaml

**Comment by xabbuh** [comment]
Can you give the name of the class for which the error in your description occurred?

**Comment by damiarita** [comment]
App\Entity\BOE\User

**Comment by xabbuh** [comment]
Can you create a small example application that allows to reproduce this?

**Comment by damiarita** [comment]
These are the commands to see the error:
`composer create-project symfony/skeleton test`
`cd test`
`composer require symfony/orm-pack`
`composer require doctrine/doctrine-migrations-bundle`
`composer require symfony/maker-bundle --dev`

Now, in `config\packages\doctrine.yaml`
`parameters:
    # Adds a fallback DATABASE_URL if the env var is not set.
    # This allows you to run cache:warmup even if your
    # environment variables are not available yet.
    # You should not need to change this value.
    env(DATABASE_URL): ''
    env(BOE_DATABASE_DRIVER): 'pdo_mysql'
    env(BOE_DATABASE_VERSION): '5.6.34'
    env(BOE_DATABASE_CHARSET): 'utf8mb4'
    env(BOE_DATABASE_URL): ''

doctrine:
    dbal:
        default_connection: BOE
        connections:
            BOE:
                driver: '%env(BOE_DATABASE_DRIVER)%'
                server_version: '%env(BOE_DATABASE_VERSION)%'
                charset: '%env(BOE_DATABASE_CHARSET)%'
                url: '%env(BOE_DATABASE_URL)%'
            BAS:
                driver: 'pdo_mysql'
                server_version: '5.6.34'
                charset: utf8mb4
                url: '%env(DATABASE_URL)%'
    orm:
        default_entity_manager: BOE
        entity_managers:
            BOE:
                connection: BOE
                mappings:
                    BOE:
                        is_bundle: false
                        type: annotation
                        dir: '%kernel.project_dir%/src/Entity/BOE'
                        prefix: 'App\Entity\BOE'
                        alias: BOE
            BAS:
                connection: BAS
                mappings:
                    BAS:
                        is_bundle: false
                        type: annotation
                        dir: '%kernel.project_dir%/src/Entity/BAS'
                        prefix: 'App\Entity\BAS'
                        alias: BAS
        auto_generate_proxy_classes: '%kernel.debug%'`

In `.env` add (replacing the BOE_DATABASE_URL url witht the right params and runnin a database there):
`
BOE_DATABASE_URL=mysql://user:pw@localhost:3306/test
BOE_DATABASE_CHARSET=utf8mb4
BOE_DATABASE_VERSION=5.5.62
BOE_DATABASE_DRIVER=pdo_mysql
`

In `src\Controller\testConstroller.php`

`<?php

namespace App\Controller;

use Symfony\Component\Routing\Annotation\Route;
use Symfony\Component\HttpFoundation\JsonResponse;
use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
use App\Entity\BOE\myEntity;


class testConstroller extends AbstractController
{
    
    /**
     * @Route ("/test/",  name="test")
     */
    public function ajaxGuideDepartures(): JsonResponse
    {
        $instance = new myEntity;
        $instance->setName('Test');
        
        $entityManager = $this->getDoctrine()->getManager();
        $entityManager->persist($instance);
        $entityManager->flush();

        return new JsonResponse($instance);

    }

}`

Finally, we create the entity (calledBOE\myEntity with a field called name as string):
`php bin/console make:entity`


We test in debug:
`php bin/console cache:clear`
`cd public`
`php -S localhost:8000`
We go with the browser to localhost:8000/test/ and it prints {} + database has a new entry

We test in prod. Change APP_ENV in .env to prod:
`cd ..`
`php bin/console cache:clear`
`cd public`
`php -S localhost:8000`
We go with the browser to localhost:8000/test/ and it prints an error

**Comment by xabbuh** [comment]
Thank you for providing the steps to reproduce. You need to tweak the config a bit to tell Doctrine the default entity manager for the `prod` environment:

```yaml
# config/packages/prod/doctrine.yaml
doctrine:
    orm:
        default_entity_manager: BOE
        auto_generate_proxy_classes: false
        metadata_cache_driver:
            type: service
            id: doctrine.system_cache_provider
        query_cache_driver:
            type: service
            id: doctrine.system_cache_provider
        result_cache_driver:
            type: service
            id: doctrine.result_cache_provider

# ...
```

When using the names used in the docs article, things work as expected since `default` is used internally by Doctrine by default (see also doctrine/DoctrineBundle#816).

**Comment by xabbuh** [comment]
Can you check if #10827 makes things more clear to you?

**Comment by damiarita** [comment]
It looks a lot better.

Could we not add BOE as an example in the docs? Could you replace it with myDefaultManager or something similar?

Thanks

**Comment by xabbuh** [comment]
Oh of course, that was a bad copy and paste from the example application. Thanks for catching it.

**Comment by javiereguiluz** [comment]
Closing as fixed by #10827.
